include/geometric_shapes/bodies.h

```diff
@@ -37,7 +37,7 @@
 #ifndef GEOMETRIC_SHAPES_BODIES_
 #define GEOMETRIC_SHAPES_BODIES_
 
-#if __cplusplus <= 199711L
+#if __cplusplus <= 199711L && _MSVC_LANG <= 199711L
```

Comment:
Shouldn't this be `||`?

Reply:
It should `&&` because we don't want to enter the if. On windows, the first condition is always true due to the incorrect value of `__cplusplus`, so with `||` it would always the enter the if and give an error.

Reply:
Makes sense... On second thought, these checks and errors seem redundant to me. I don't think it's realistic that someone uses C++98 with ROS2. @tylerjw what do you think?

Reply:
I don't think that's reasonable considering that ROS1 itself standardizes on C++11 and ROS2 on C++14.

Reply:
So let's remove these checks I'd say

Reply:
Are we maybe not telling the compiler we want C++11 or newer somehow and MSVCC defaults to C++98?

Reply:
Even though the compiler is for C++11 and newer, the macro value is incorrect for MSVC without the `Zc:__cplusplus` compiler flag. This [Microsoft blog post](https://devblogs.microsoft.com/cppblog/msvc-now-correctly-reports-__cplusplus/) explains this in more detail.

Reply:
I read the blog post and I'm still confused.  Is this just a way to detect having an old msvcc compiler based on them taking longer to update this macro?

Reply:
Even on newer MSVC compilers the macro is still wrong. The only way to tell the compiler to use the correct value is through the use of the compiler flag. I didn't just add that compiler flag to the compile options because that would require all packages that include those header files to also add that compiler option. Instead, MSVC uses a different macro (`_MSVC_LANG`) that has the same value as `__cplusplus` on other compilers. And this doesn't need any compiler flag to be accurate.

However, based on the discussion above, I thought I should remove those checks. If not, I can revert that commit.

Reply:
Ok, I think I understand now.  Thank you.  I think removing those checks is just fine as ROS doesn't support old C++ standards so we shouldn't need to check for them anyway.